File: 3zadanie.cpp
#include <iostream>

int digitSum(int x) {
    int sum = 0;
    while (x > 0) {
        sum += x % 10;
        x /= 10;
    }
    return sum;
}

int main() {
    int N;
    std::cout << "Введите количество символов N: ";
    std::cin >> N;

    if (!std::cin || N <= 0) {
        std::cout << "Ошибка ввода";
        return 0;
    }

    int count = 0;

    for (int i = 0; i < N; i++) {
        int x;
        std::cin >> x;

        if (!std::cin || x <= 0) {
            std::cout << "Ошибка ввода";
            return 0;
        }

        if (digitSum(x) > 10) {
            count++;
        }
    }

    std::cout << "Количество чисел у которых сумма цифр больше 10: " << count << std::endl;
    return 0;
}

File: 3zadanie.go
package main

import "fmt"

func digitSum(x int) int {
	sum := 0
	for x > 0 {
		sum += x % 10
		x /= 10
	}
	return sum
}

func main() {
	var N int

	fmt.Print("Введите количество символов N: ")
	if _, err := fmt.Scan(&N); err != nil || N <= 0 {
		fmt.Print("Ошибка ввода")
		return
	}

	count := 0

	for i := 0; i < N; i++ {
		var x int
		if _, err := fmt.Scan(&x); err != nil || x <= 0 {
			fmt.Print("Ошибка ввода")
			return
		}

		if digitSum(x) > 10 {
			count++
		}
	}

	fmt.Println("Количество чисел у которых сумма цифр больше 10:", count)
}